docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+
+**Versioning Scheme**: [Kubernetes](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) (e.g. v1alpha1, v1beta2, v1, v2, etc)
+
+**Update Process**: 
+
+   1) Add a new folder in the source repository (`pkg/apis/workspaces` in `devfile/api`)
+   2) Add a schema and version in the CRD manifests
+   3) Generate the JSON schema from the API. New JSON schema will be located under `schemas/latest`.
+   4) Update the devworkspace operator and devfile library to consume the Go structs in the new API version, as needed.
+
+**When to Update?**  As incrementing the Kubernetes API version for Devfile is a relatively heavy process, and affects the library, only update the K8s API version when absolutelyneeded 
+   - for **big** changes or backwards incompatible changes.
+
+### Devfile JSON Schema
+
+As mentioned above, the Devfile JSON schema is generated from the Go structs defined in the Devfile Kubernetes API. The latest JSON schema for a given API version is located under `schemas/<api-version>` in the [devfile/api repo](https://github.com/devfile/api/). 
+
+**Versioning Scheme**: Semver (major.minor.bugfix)
+
+**Update Process**: 
+  
+   1) Update the schema version string in the `// +devfile:jsonschema:version=<schema-version>` annotation in `pkg/apis/workspace/<k8s-api-version>/doc.go`
+   2) Re-generate the json schema
+   3) Publish new schema to devfile.io website
+
+**When to Update?** On each release of the schema, incremented based on the changes going in to the release (e.g. bugfix = comments / bugfixes, minor = larger and / or backwards compatible changes, major == breaking / backwards incompatible changes). K8S API version updates should also result in an appropriate increment of the schema version.
+
+
+### Relationship Between K8s API version and JSON Schema Version
+
+The Devfile JSON schema is generated from the Kubernetes API, and the version for the JSON schema is set in the doc.go file in the K8S API (`pkg/apis/workspace/<api-version>/doc.go).
+
+As we’re only updating the K8S API version when needed, but incrementing the schema version more frequently, this means that any given API version may point to multiple, backwards-compatible, schema versions over its lifespan. The schema version under `schemas/<api-version>` in [devfile/api repo](https://github.com/devfile/api/) points to the latest JSON schema generated from the K8S API.
+
+## Release Process
+The following steps outline the steps done to release a new version of the Devfile schema and publish its schemas to the devfile.io website
+
+   1) The release engineer tasked with creating the release clones the repository (and checks out the release branch if one already exists)
+
+   2) `make-release.sh <schema-version> <k8s-api-version>` is run:
+
+      i) A release branch (the name corresponding to the schema version) is created, if one does not already exist.
```

Comment:
Does the release branch correspond to the z-stream version or the minor revision? I.e. are the release branches `v2.1.x` or `v2.1.0`, `v2.1.1`, etc.?

Reply:
Release branches correspond to the minor version, so `v{major}.{minor}.x`